routes/index.js

```diff
@@ -7,7 +7,17 @@ const parseRoutes = require('./parseRoutes')
 
 
     , server = (serverType, routesJson, config) => {
-        return serverType.createServer(require('./router.js')(routesJson, config));
+        let serverReturn;
+
+        if (config.key && config.cert) {
+            serverReturn = serverType.createServer({
```

Comment:
The convention is pretty solid. All the libraries I've found for http2/spdy is a createServer function with an options object as the first parameter. Which is the same signature and function used by the built in http/https modules.

Based on that I think that documentation might be nice but is not a must have.

-Daniel

> On Nov 29, 2015, at 10:45 AM, Cory <notifications@github.com> wrote:
> 
> In routes/index.js:
> 
> > @@ -7,7 +7,17 @@ const parseRoutes = require('./parseRoutes')
> >  
> >  
> >      , server = (serverType, routesJson, config) => {
> > -        return serverType.createServer(require('./router.js')(routesJson, config));
> > +        let serverReturn;
> > +
> > +        if (config.key && config.cert) {
> > +            serverReturn = serverType.createServer({
> Is it needful to document the interface the httpServer should have or is there sufficient convention around Node httpServers (http/https,http2, etc) to have a createServer method for, uh, creating the server :)
> 
> —
> Reply to this email directly or view it on GitHub.
>